tests/test_api/test_item.py

```diff
@@ -19,6 +26,38 @@ def test_get_item_missing(self, test_client, item):
         check_valid_header_type(r.headers)
         assert r.status_code == 404
 
+    def test_put_item_upvote(self, test_client, user, item):
+        self.login(test_client, user.email, user.password)
+
+        upvote = dict(vote=1)
```

Comment:
Let's do integer and use your comment above with checking whether a user should be able to keep voting.

Also, disabling buttons on the frontend once you've voted is a good way to stop it from the user side.